include/nova_renderer/shaderpack_data.hpp

```diff
@@ -9,17 +9,21 @@
 #define NOVA_RENDERER_SHADERPACK_DATA_HPP
 
 #include <cstdint>
-#include <nlohmann/json.hpp>
+#include <functional>
```

Comment:
While this is not a problem with this PR, I would like to point out, that we should look into replacing `std::function` because of the uncontrollable allocation it does. Should probably be further discussed on discord.

Reply:
I think we need to get Nova using custom allocators throughout before we make our own version of `std::function`, but yes I absolutely agree that it needs to happen